### DISCUSSION TITLE
$root and data-message

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#3889] [4. General]
![image](https://github.com/alpinejs/alpine/assets/9047436/6b55ee18-f69d-4fe1-bd2e-b6aee2ddced5)

Hi,

I am not clear about the data-message meant here, I cannot find any doc about this usage. That does that mean?

Thanks
Eric Xin

**Comment by ekwoka** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset

This should clear that up :)

It's not an Alpine thing.

this here is just demoing that `$root` is the ELEMENT that the nearest `x-data` is on, not some alpine special object.

**Comment by ericxin1982** [comment]
Thanks so much!
